### PR TITLE
For #7356: Scroll to selected tab when pressing tab overview 

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### After merge
- [ ] **Milestone**: Make sure issues finished by this pull request are added to the [milestone](https://github.com/mozilla-mobile/fenix/milestones) of the version currently in development.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarController.kt [hunk 1]
```diff
@@ -302,14 +302,10 @@ class DefaultBrowserToolbarController(
...
             delay(ANIMATION_DELAY)
-            if (!navController.popBackStack(R.id.homeFragment, false)) {
-                val directions = BrowserFragmentDirections.actionBrowserFragmentToHomeFragment()
-                navController.nav(
-                    R.id.browserFragment,
-                    directions,
-                    null
-                )
-            }
+            val directions = BrowserFragmentDirections.actionBrowserFragmentToHomeFragment(
```
Comment (BranescuMihai) [inline app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarController.kt]: You used the old type of action, I assume you wanted to use the one with `popUpTo` right?

File: app/src/main/res/navigation/nav_graph.xml [hunk 2]
```diff
@@ -184,6 +188,10 @@
         <action
             android:id="@+id/action_browserFragment_to_homeFragment"
             app:destination="@id/homeFragment" />
+        <action
```
Comment (BranescuMihai) [inline app/src/main/res/navigation/nav_graph.xml]: Maybe a bit more suggestive name like action_browserFragment_to_homeFragment_popUp?

Also, is the old action still needed? If not, maybe we could remove it.

**Comment by ekager** [comment]
cc @sblatz my instinct says no, but wanted to check if you had concerns with this and the animation work you've been doing?

**Comment by liuche** [comment]
Haven't seen any activity here so I think this is fine to merge. We can be on the lookout for any bad interactions with the tab.

**Pull request merged**